src/database/migrations/20240422051324-create-role.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const { UUIDV4 } = require('sequelize');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('Roles', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.UUID,
+        defaultValue: UUIDV4,
+      },
+      name: {
+        type: Sequelize.STRING,
+      },
```

Comment:
That's good and impressive work, I would like to highly recommend you to consider that each role has it's specific permissions, which it needs to be assigned to, and in addition, there should be a sufficient relationship between each user and role... kindly consider those recommendations, it will enhance your PR and make it comprehensive

Reply:
Yeah Thanks for this suggestion. But actually the permission feature is a task on its own. I guess I will have to push them separately. I will have to create it's unique branch that blends in all of that.